File: src/main/java/com/codingforcookies/betterrecords/common/packets/PacketSoundStop.java
package com.codingforcookies.betterrecords.common.packets;

import io.netty.buffer.ByteBuf;
import net.minecraft.entity.player.EntityPlayer;

import com.codingforcookies.betterrecords.client.sound.SoundHandler;

import net.minecraftforge.fml.common.network.ByteBufUtils;

public class PacketSoundStop implements IPacket {
	int x, y, z, dimension;
	
	public PacketSoundStop() { }
	
	public PacketSoundStop(int x, int y, int z, int dimension) {
		this.x = x;
		this.y = y;
		this.z = z;
		this.dimension = dimension;
	}
	
	public void readBytes(ByteBuf bytes) {
		String recieved = ByteBufUtils.readUTF8String(bytes);
		String[] str = recieved.split("\\]");
		x = Integer.parseInt(str[0]);
		y = Integer.parseInt(str[1]);
		z = Integer.parseInt(str[2]);
		dimension = Integer.parseInt(str[3]);
	}
	
	public void writeBytes(ByteBuf bytes) {
		ByteBufUtils.writeUTF8String(bytes, x + "]" + y + "]" + z + "]" + dimension);
	}
	
	public void executeClient(EntityPlayer player) {
		if(SoundHandler.soundPlaying.containsKey(x + "," + y + "," + z + "," + dimension))
			SoundHandler.soundPlaying.remove(x + "," + y + "," + z + "," + dimension);
	}
	
	public void executeServer(EntityPlayer player) { }
}

File: src/main/kotlin/com/codingforcookies/betterrecords/network/PacketSoundStop.kt
package com.codingforcookies.betterrecords.network

import com.codingforcookies.betterrecords.client.sound.SoundHandler
import io.netty.buffer.ByteBuf
import net.minecraft.util.math.BlockPos
import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler
import net.minecraftforge.fml.common.network.simpleimpl.MessageContext

class PacketSoundStop @JvmOverloads constructor(
        var pos: BlockPos = BlockPos(0, 0, 0),
        var dimension: Int = -1
) : IMessage {

    override fun toBytes(buf: ByteBuf) {
        buf.writeInt(pos.x)
        buf.writeInt(pos.y)
        buf.writeInt(pos.z)

        buf.writeInt(dimension)
    }

    override fun fromBytes(buf: ByteBuf) {
        pos = BlockPos(buf.readInt(), buf.readInt(), buf.readInt())

        dimension = buf.readInt()
    }

    class Handler : IMessageHandler<PacketSoundStop, IMessage> {

        override fun onMessage(message: PacketSoundStop, ctx: MessageContext): IMessage? {
            with(message) {
                SoundHandler.stopPlaying(pos, dimension)
            }

            return null
        }
    }
}